src/lib/core/cord_on_demand.cc

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2022, Tarantool AUTHORS, please see AUTHORS file.
+ */
+
+#include "cord_on_demand.h"
+
+#include <stdlib.h>
+
+#include "fiber.h"
+#include "trivia/util.h"
+
+/**
+ * RAII wrapper around a cord object.
+ *
+ * A thread-local cord object is created on the first call to the get() method
+ * and destroyed when the thread exits.
+ */
+class CordOnDemand final {
```

Comment:
Nit: this class doesn't really provide any encapsulation — in this case I usually use `struct` declaration.

Reply:
I'm following the [Google C++ style guide](https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes) here:

> Use a struct only for passive objects that carry data; everything else is a class.

Reply:
If we would decide to discard the `get` method, this object would actually only carry data (static methods don’t account here, AFAIC).